src/rest_api/classes/gene/widgets/interactions.clj

```diff
@@ -91,16 +54,67 @@
      (not
       [?ix :interaction/type :interaction.type/predicted])
      [?ih :interaction.interactor-overlapping-gene/gene ?neighbour]
-     [(not= ?gene ?neighbour)]]
+     [(not= ?gene ?neighbour)]]])
+
+(defn interactor-idents [db]
+  (sort (d/q '[:find [?ident ...]
+               :where
+               [?e :db/ident ?ident]
+               [?e :pace/use-ns "interactor-info"]
+               [_ :db/valueType :db.type/ref]
+               [_ :db.install/attribute ?e]
+               [(namespace ?ident) ?ns]
+               [(= ?ns "interaction")]]
+             db)))
 
-    ;; Not actually used since count trick is subtantially faster.
-    [(gene-neighbour-interaction ?gene ?int)
-     (gene-neighbour ?gene ?n1)
-     (gene-interaction ?n1 ?int)
-     [?int :interaction/interactor-overlapping-gene ?ih]
-     [?ih :interaction.interactor-overlapping-gene/gene ?n2]
-     [(not= ?n1 ?n2)]
-     (gene-neighbour ?gene ?n2)]])
+;; Schema doesn't change once a process is running
+(def interactor-refs (memoize interactor-idents))
+
+(defn- interactor-role [interactor]
+  (let [int-types (:interactor-info/interactor-type interactor)]
+    (or (interactor-role-map (first int-types))
+        (if (corresponding-cds (interaction-target interactor))
+          :associated-product)
+        :other)))
+
+(defn- regulatory-result [interaction]
+  (some->> (:interaction/regulation-result interaction)
+           (map :interaction.regulation-result/value)
+           first))
+
+(defn- humanize-name [ident]
+  (let [ident-name (name ident)
+        hname (-> (name ident)
+                  (str/split #":")
+                  (last)
+                  (str/replace #"-" " ")
+                  (str/capitalize))]
+    ;; hacks for when hname trick isn't good enough
+    (cond
+      (= hname "Proteinprotein") "ProteinProtein"
+      :default hname)))
+
+(defn- interaction-type-name [interaction]
+  (let [itype (first (:interaction/type interaction))
+        type-name (humanize-name itype)]
+    (cond
+      ;; Hack to produce the "type-name" when real type-name
+      ;; is regulatory.
+      ;; TODO: Perhaps the proposed module system should be able to address
+      ;;       this and produce a cleaner solution.
+      (str/includes? (str/lower-case type-name) "regulat")
+      (if-let [reg-res (regulatory-result interaction)]
```

Comment:
I have started using when-let instead of if-let when possible. The difference is that when-let does not have an "else" part. This makes it a bit more readable because you know that there is no "else" to the condition.